### DISCUSSION TITLE
Problem with Adam optimizer while using MLPRegressor.

### DISCUSSION THREAD

**Discussion by tamoghnabh** [online-ml/river#714] [Q&A] [answered]
I am trying to fit a regression model using MLPRegressor module. It works fine with SGD as the optimizer. But I get the following error `'numpy.ndarray' object has no attribute 'items'` if I use Adam as the optimizer. Does this problem has something to do with the statement in the API reference stating "This model is still work in progress." Is there any workaround as I am expecting improvement in performance with Adam?

**Comment by MaxHalford** [comment]
Indeed not all of the optimizers can handle both numpy arrays and dicts. @gbolmier had started to adapt the optimizers in #338, but it never got finished.

**Comment by MaxHalford** [comment, marked as answer]
@Geronimo1995 we're adding support for Adam in #722

**Reply by tamoghnabh** [reply]
That is fantastic news. Expecting significant performance improvement in my model.

**Reply by MaxHalford** [reply]
It's merged. You can install the latest development version to use it.

**Reply by tamoghnabh** [reply]
It seems that the problem persists even after updating to the latest version.

**Reply by MaxHalford** [reply]
Ok I found the issue and I just fixed it. I did a dirty hack allowing each layer to have its own optimizer. TBD if it's the best way to go. Try again and it should work!

**Reply by tamoghnabh** [reply]
Worked like a CHARM!! Thank you so much.
